Sources/Commands/SwiftTool.swift

```diff
@@ -493,8 +493,15 @@ public class SwiftTool {
     }
 
     func getAuthorizationProvider() throws -> AuthorizationProvider? {
-        // currently only single provider: netrc
-        return try self.getNetrcConfig()?.get()
+        var providers = [AuthorizationProvider]()
+        // netrc file has higher specificity than keychain so use it first
+        if let workspaceNetrc = try self.getNetrcConfig()?.get() {
+            providers.append(workspaceNetrc)
+        }
+#if canImport(Security)
+        providers.append(KeychainAuthorizationProvider())
```

Comment:
We might consider adding a `--no-keychain` flag to allow opt-out.

Reply:
I added a TODO and commented this out for now to avoid changing current behavior. I will create follow-up PR to consolidate auth logic.